tls/s2n_quic_support.c

```diff
@@ -37,9 +37,6 @@ S2N_RESULT s2n_read_in_bytes(struct s2n_connection *conn, struct s2n_stuffer *ou
 
 int s2n_connection_enable_quic(struct s2n_connection *conn)
 {
-    /* The QUIC protocol doesn't use pre-1.3 TLS */
-    ENSURE_POSIX(s2n_is_tls13_enabled(), S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED);
```

Comment:
Why wasn't this replaced with a `s2n_connection_get_protocol_version(conn)` check like the others?

Reply:
QUIC is enabled before the protocol version is negotiated, so we don't know whether it's really TLS1.2 or TLS1.3 at this point. This check was never complete, since a connection can negotiate TLS1.2 even if TLS1.3 is supported.

However, there are other checks to make sure we actually negotiate TLS1.3 with QUIC: [Server](https://github.com/awslabs/s2n/blob/88e76f3df167c437cf88758b9bb07062711c5ec2/tls/s2n_client_hello.c#L263) and [Client](https://github.com/awslabs/s2n/blob/e6314dd844c50a90f3a489a177f1ff479bebe7b7/tls/s2n_server_hello.c#L143)